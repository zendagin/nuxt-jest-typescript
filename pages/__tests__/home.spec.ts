import { shallowMount } from "@vue/test-utils";
import Home from "../index.vue";


const factory = () => {
    return shallowMount(Home, {
    });
};

describe("Home", () => {
    test("mounts properly", () => {
        const wrapper = factory();
        expect(wrapper.vm).toBeTruthy();
    });

    test("renders properly", () => {
        const wrapper = factory();
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("title", () => {
        const wrapper = factory();
        const title = wrapper.find(".title")
        expect(title.text()).toBe("nuxt-jest");
    });
});
