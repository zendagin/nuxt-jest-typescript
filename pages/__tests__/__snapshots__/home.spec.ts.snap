// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly 1`] = `
<section class="container">
  <div>
    <logo-stub></logo-stub>
    <h1 class="title">
      nuxt-jest
    </h1>
    <h2 class="subtitle">
      Nuxt jest sample config
    </h2>
    <div class="links"><a href="https://nuxtjs.org/" target="_blank" class="button--green">Documentation</a> <a href="https://github.com/nuxt/nuxt.js" target="_blank" class="button--grey">GitHub</a></div>
  </div>
</section>
`;
